Source/Core/InputCommon/ControllerInterface/DInput/DInputKeyboardMouse.cpp

```diff
@@ -156,38 +160,106 @@ KeyboardMouse::KeyboardMouse(const LPDIRECTINPUTDEVICE8 kb_device,
   for (unsigned int i = 0; i != mouse_caps.dwAxes; ++i)
   {
-    AddInput(new RelativeMouseAxis(i, false, &m_state_in.relative_mouse));
-    AddInput(new RelativeMouseAxis(i, true, &m_state_in.relative_mouse));
+    AddInput(new RelativeMouseAxis(i, false, &m_current_state.relative_mouse));
+    AddInput(new RelativeMouseAxis(i, true, &m_current_state.relative_mouse));
   }
 }
 
-void KeyboardMouse::UpdateCursorInput()
+std::pair<double, double> CalculateFinalCursorValue(const double& window_width,
+                                                    const double& window_height,
+                                                    const double& sensitivity,
+                                                    const double& window_ratio,
+                                                    const Point& raw_mouse_point)
 {
-  POINT point = {};
-  GetCursorPos(&point);
+  std::pair<double, double> return_pair;
+  return_pair.first = ((raw_mouse_point.x / window_width) - 0.5) * sensitivity * window_ratio;
+  return_pair.second = ((raw_mouse_point.y / window_height) - 0.5) * sensitivity;
+  return return_pair;
+}
 
-  // Get the cursor position relative to the upper left corner of the current window
-  // (separate or render to main)
-  ScreenToClient(s_hwnd, &point);
+void KeyboardMouse::UpdateCursorInput()
+{
+  POINT raw_mouse_point = {0, 0};
+  GetCursorPos(&raw_mouse_point);
+
+  ExtendedWindowInfo render_window_info =
+      OctagonalMouseJail::GetInstance().GetExtendedRenderWindowInfo();
+  double sensitivity = Config::Get(Config::MOUSE_SENSITIVITY) *
+                       Config::scaling_factor_to_make_mouse_sensitivity_one_for_fullscreen;
+  ;
+  bool jail_enabled = Config::Get(Config::MOUSE_OCTAGONAL_JAIL_ENABLED);
+  const Common::Vec2 window_scale = g_controller_interface.GetWindowInputScale();
+  // Sets the cursor to the center of the screen during loading to avoid
+  // drift(when booting straight to game). This code will result in drift if boot to pause is used
+  // with the OctagonalMouseJail.
+  if ((OctagonalMouseJail::GetInstance().m_player_gets_center && Host_RendererHasFocus()) ||
+      (::Core::GetState() == ::Core::State::Starting && Host_RendererHasFocus() && jail_enabled))
+  {
+    SetCursorPos(static_cast<long>(render_window_info.center.x),
+                 static_cast<long>(render_window_info.center.y));
+    OctagonalMouseJail::GetInstance().m_player_gets_center = false;
+    return;
+  }
 
-  // Get the size of the current window (in my case Rect.top and Rect.left was zero).
-  RECT rect;
-  GetClientRect(s_hwnd, &rect);
+  if (render_window_info.handle == nullptr)
+  {
+    ScreenToClient(s_hwnd, &raw_mouse_point);
+    Point mouse_point{static_cast<double>(raw_mouse_point.x),
+                      static_cast<double>(raw_mouse_point.y)};
+    WINDOWINFO current_window_info{};
+    current_window_info.cbSize = sizeof(WINDOWINFO);
+    GetWindowInfo(s_hwnd, &current_window_info);
```

Comment:
Is there a reason you're using `GetWindowInfo()` instead of the `GetClientRect()` that was there previously? Client area seems more appropriate to use here, since window decorations (the title bar and the window delimiter lines on the other edges) probably shouldn't affect this calculation.

Reply:
Nope, just didn't know you could do that. I don't have the old version pulled up when I'm writing, but I may try to find a way to add those diff lines into my code. The last time I looked at the old code was 6 months ago.